src/main/java/hudson/plugins/jobConfigHistory/ConfigInfoCollector.java

```diff
@@ -159,4 +159,30 @@ void getConfigsForType(File itemDir, String folderName) throws IOException {
 		}
 		return configs;
 	}
+
+	public List<ConfigInfo> collect() throws  IOException{
+		final File[] itemDirs;
+		if ("deleted".equals(type)) {
+			itemDirs = overViewhistoryDao.getDeletedJobs();
+		} else {
+			itemDirs = (File[]) ArrayUtils.addAll(
+					overViewhistoryDao.getDeletedJobs(),
+					overViewhistoryDao.getJobs());
+		}
+		Arrays.sort(itemDirs, FileNameComparator.INSTANCE);
+		for (final File itemDir : itemDirs) {
+			String folderName = getFolderName(itemDir);
+			//System.out.println("itemDir: " + itemDir + ",\n  folderName: " + folderName);
+			getConfigsForType(itemDir, folderName);
+		}
+		return configs;
+	}
+
+	private String getFolderName(File file) {
+		//too damn hacky...
+		if (file.getParentFile().getName().equals("jobs")
+				&& file.getParentFile().getParentFile().getParentFile().getName().equals("jobs")) {

```

Comment:
Does this work in all cases? Note: There can be folders in folders in folder etc.. I think there might be an easier way.

Reply:
Just tested: Create a folder job called "FolderA" and within this folder another folder job called "FolderB". Then create a classical (e.g. freestyle) job within this nested folder. Delete this job and you will see, that the deletion of this job still doesn't get shown.

Reply:
Note: A "Folder" ist not just a directory on the file system. It is a special job type, that can't get built directly. See: https://plugins.jenkins.io/cloudbees-folder

Reply:
Hint: You might add `cloudbees-folder` as optional dependency to the `pom.xml`.

Reply:
This should be working now. Thanks for the hint!